### DISCUSSION TITLE
How to filter in like with lnav?

### DISCUSSION THREAD

**Discussion by piotr-dobrogost** [krasa/GrepConsole#308] [General]
How can I filter in lines containing some string (show only lines containing the string) like with lnav.org?
I thought putting the string in _Unless expression_ and setting _Action_ to _REMOVE_ would do the trick but it does nothing.

**Comment by krasa** [comment]
Excluding all but `FSRecords`:
![unnamed](https://github.com/user-attachments/assets/03fc4d56-d743-42ad-98a2-785714a1cdec)

or

filtering all but `Action`
![image](https://github.com/user-attachments/assets/a4235121-f476-4322-8155-9adb62141b03)
 - `whole line` must be turned on

**Reply by piotr-dobrogost** [reply]
Thanks for the answer.
Both variants do what I was looking for.
Not sure how do they differ (if at all)?
By the way, after going to _Open Grep Console settings_ and changing settings, the new settings do not take effect until after I try searching for something using box in the upper right corner:
![Screenshot_20240718_120543](https://github.com/user-attachments/assets/20175be1-f847-4cf0-b162-1cf319cf70de)
which is not intuitive.

**Comment by krasa** [comment]
> new settings do not take effect 

it's just that the console does not get reloaded. It is not feasible to do that.


Maybe Grep would be more suitable
![image](https://github.com/user-attachments/assets/a6af9455-e31c-4674-8151-b497f4493315)

**Reply by piotr-dobrogost** [reply]
Grep looks good, thanks for bringing my attention to it.
